iocore/utils/PropertyBlock.h

```diff
@@ -0,0 +1,254 @@
+#pragma once
+#ifndef _PropertyBlock_h
+#define _PropertyBlock_h
+
+#include "stdint.h"
+#include <cstddef>
+#include <atomic>
+#include <cstring>
+#include <vector>
+#include <type_traits>
+
+#include "ts/ink_assert.h"
+
+/// runtime defined static container
+/** The size of this structure is actually zero, so it will not change the size of your derived class.
+ * But new and delete are overriden to use allocate enough bytes of the derived type + properties added
+ * at run time using ioBufferAllocator.
+ *
+ * Also all bools are packed to save space using the *Bit methods.
+ *
+ * This is templated so static variables are instanced per Derived type. B/c we need to have different
+ * size property sets.
+ */
+template <class Derived_t> ///< static members instance pey type.
+struct PropertyBlock {
+public:
+  using self_t = PropertyBlock<Derived_t>;
```

Comment:
Jason has forced us all to use `self_type` for this.

Reply:
it's unused. I'll remove it.